dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/alias/InclusionBasedPointerAnalysis.java

```diff
@@ -155,11 +158,13 @@ private void run(Program program) {
             //TODO replace with removeFirst() when using java 21 or newer
             final Variable variable = queue.keySet().iterator().next();
             final List<Offset<Variable>> edges = queue.remove(variable);
+            logger.trace("dequeue {}", variable);
             algorithm(variable, edges);
         }
-        //printGraph();
+        printGraph();
+        final SyntacticContextAnalysis synContext = SyntacticContextAnalysis.newInstance(program);
         for (final Map.Entry<MemoryCoreEvent, Offset<Variable>> entry : eventAddressSpaceMap.entrySet()) {
-            postProcess(entry);
+            postProcess(entry, synContext);
```

Comment:
Since you use the syntactic context only inside `postProcess`, I think it would be better to pass the `program` instance and inside `postProcess ` do this:
```
final SyntacticContextAnalysis synContext = logger.isWarnEnabled ? SyntacticContextAnalysis.newInstance(program) : null;
```
IIRC, `SyntacticContextAnalysis` computes quite a lot of data (a whole context stack per event) so it might be better to avoid it if unused. Realistically speaking, the analysis' time/memory consumption won't be noticable though.

Reply:
Usually, `logger.isWarnEnabled()` would be enabled either way.  I attempt to use `Suppliers.memoize(Supplier)`.  That way, at least one error message has to be issued for invoking `SyntacticContextAnalysis`.  And it would be done at most once.

Reply:
I like that solution